### PR TITLE
Python package set major updates 

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by FRidh** [comment]
@jonringer feel free to force-push.

**Comment by FRidh** [comment]
Should also consider https://github.com/NixOS/nixpkgs/pull/65023 again

**Comment by marsam** [comment]
can you revert the gym update https://github.com/NixOS/nixpkgs/pull/76059/commits/834a4f76d2aa951f93faa41e9980352f3327be9e?, because it requires python-opencv which is not in nixpkgs yet

**Comment by FRidh** [comment]
Done

**Comment by FRidh** [comment]
Could you refer me to that `python-opencv`. It's a mess with all those different `opencv` packages.

**Comment by marsam** [comment]
Thanks.
It's https://pypi.org/project/opencv-python/ https://github.com/skvark/opencv-python 
I recall @jonringer mentioning that packaging it was not a trivial task

**Comment by jonringer** [comment]
IIRC, we can also use the opencv package to make the bindings, the https://github.com/skvark/opencv-python repo is just there to publish the wheels.

But tying to build from opencv-python is painful as it makes a lot of assumptions about network access, git, and cmake integration.

**Comment by FRidh** [comment]
Yes, we just need to build a dist-info folder that has the `opencv-python` name.

**Comment by jonringer** [comment]
the only packages i can think of that use opencv-python are imgaug, baselines, and gym

**Comment by jonringer** [comment]
there's a PR to fix imgaug here: https://github.com/NixOS/nixpkgs/pull/67494
gym doesn't need it unless it's bumped
and baselines currently removes it from the install_requires

Although eventually, we should add it. I just don't have the will right now.

**Comment by jonringer** [comment]
did you want help stabilizing this? because I know a lot of leaf packages probably got broken

**Comment by FRidh** [comment]
Yes, I think we need to do such an upgrade again. I saw some of the PR's, and those are bound to cause regressions as well. With this PR/branch we can test them.

I will actually be away for a week so probably need to rerun this after.

**Comment by jonringer** [comment]
I feel like now would be a good time to bump pytest and hypothesis

**Comment by FRidh** [comment]
> I feel like now would be a good time to bump pytest and hypothesis

Definitely!

Thank you for looking at these packages. Next Sunday I'll be back and can have a look as well.

**Comment by jonringer** [comment]
had to revert sip bump, as it broke a lot of pyqt builds

**Comment by jonringer** [comment]
going to revert the boto/moto bumps apparently aws is a very fragile api

**Comment by jonringer** [comment]
also reverted the major coverage bump, as that broke quite a few packages

**Comment by jonringer** [comment]
did you want to bump python3.{6,7,8} on this branch? @FRidh

**Comment by jonringer** [comment]
https://www.python.org/downloads/ they all got a bump on the same day

**Comment by jonringer** [comment]
python3.9a2 was also released https://www.python.org/downloads/release/python-390a2/

**Comment by jonringer** [comment]
had to revert flit major bump. they broke it into a flit and flit_core package, but they both require flit to build a wheel

**Comment by jonringer** [comment]
reverted pytest-astropy, added a new pytest-astropy-headers dependency, which doesn't exist yet

**Comment by jonringer** [comment]
disregard the interpreter bumps, already applied

**Comment by jonringer** [comment]
starting to get to the point where the regressions are on leaf packages. I think this is ready to merge.

**Comment by jonringer** [comment]
rebased on top of staging, there shouldn't be any conficts...

**Comment by jonringer** [comment]
most of the regressions are from python38Packages.aiohttp, https://hydra.nixos.org/jobset/nixpkgs/python-unstable

**Comment by lilyball** [comment]
What's the status of this? `git-revise` is currently broken because of `importlib-metadata`.

**Comment by jonringer** [comment]
Sorry, i've been visiting family and tied of with other things for the holidays. I'm assuming @FRidh has been as well

**Comment by FRidh** [comment]
Yes indeed. I just got back so I can have a look at it again as well.

**Comment by FRidh** [comment]
Just did another rebase onto staging.

**Comment by jonringer** [comment]
I would like to see a lot of these changes in master. Not sure what to do to expedite this getting in

**Comment by FRidh** [comment]
Tomorrow I intend to merge staging next at which point this goes into the
new staging next. If there aren't many other big changes that should land
in master relatively quickly.

**Comment by FRidh** [comment]
Forgot to merge it into staging so that's why it went to staging-next...

File: pkgs/development/python-modules/gunicorn/default.nix [hunk 1]
```diff
@@ -8,11 +8,12 @@
 
 buildPythonPackage rec {
   pname = "gunicorn";
-  version = "19.9.0";
+  version = "20.0.4";
```
Comment (arnoldfarkas) [inline pkgs/development/python-modules/gunicorn/default.nix]: FYI this change breaks apache-airflow build:
```
ERROR: Could not find a version that satisfies the requirement gunicorn<20.0,>=19.5.0 (from apache-airflow==1.10.5) (from versions: none)
ERROR: No matching distribution found for gunicorn<20.0,>=19.5.0 (from apache-airflow==1.10.5)
```

Comment (jonringer) [reply pkgs/development/python-modules/gunicorn/default.nix]: opening a ticket is more action-able :), I unfortunately don't have time to look into this :(

**Pull request merged**